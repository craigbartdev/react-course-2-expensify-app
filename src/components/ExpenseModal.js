import React from 'react';
import Modal from 'react-modal';

export const ExpenseModal = (props) => (
    <Modal
        isOpen={props.modalVisible}
        onRequestClose={props.modalClose}
        contentLabel="Delete Expense"
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Are you sure that you would like to delete this expense?</h3>
        <button onClick={props.modalAction} className="modal__button remove">Yes</button>
        <button onClick={props.modalClose} className="modal__button cancel">Cancel</button>
    </Modal>
);

export default ExpenseModal;