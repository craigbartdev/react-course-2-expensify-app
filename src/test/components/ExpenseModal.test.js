import React from 'react';
import { shallow } from 'enzyme';
import ExpenseModal from '../../components/ExpenseModal';

let onRemove, modalClose, wrapper;

beforeEach(() => {
    modalClose = jest.fn();
    onRemove = jest.fn();
    wrapper = shallow(
        <ExpenseModal
            modalVisible={true}
            onRemove={onRemove}
            modalClose={modalClose}
        />
    );
});

test('should render ExpenseModal correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should test cancel button', () => {
    wrapper.find('button.remove').simulate('click');
    expect(onRemove).toHaveBeenCalled();
});

test('should test cancel button', () => {
    wrapper.find('button.cancel').simulate('click');
    expect(modalClose).toHaveBeenCalled();
});