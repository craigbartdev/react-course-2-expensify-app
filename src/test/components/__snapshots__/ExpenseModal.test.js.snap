// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Delete Expense"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <h3
    className="modal__title"
  >
    Are you sure that you would like to delete this expense?
  </h3>
  <button
    className="modal__button remove"
    onClick={[Function]}
  >
    Yes
  </button>
  <button
    className="modal__button cancel"
    onClick={[Function]}
  >
    Cancel
  </button>
</Modal>
`;
